runtime/renderer/xwalk_content_renderer_client.cc

```diff
@@ -130,6 +131,62 @@ void XWalkContentRendererClient::RenderThreadStarted() {
 #endif
 }
 
+#if defined(OS_ANDROID)
+bool XWalkContentRendererClient::HandleNavigation(
+    content::RenderFrame* render_frame,
+    bool is_content_initiated,
+    int opener_id,
+    blink::WebFrame* frame,
+    const blink::WebURLRequest& request,
+    blink::WebNavigationType type,
+    blink::WebNavigationPolicy default_policy,
+    bool is_redirect) {
+  // Only GETs can be overridden.
+  if (!request.httpMethod().equals("GET"))
+    return false;
+
+  // Any navigation from loadUrl, and goBack/Forward are considered application-
+  // initiated and hence will not yield a shouldOverrideUrlLoading() callback.
+  // Webview classic does not consider reload application-initiated so we
+  // continue the same behavior.
+  // TODO(sgurun) is_content_initiated is normally false for cross-origin
+  // navigations but since android_webview does not swap out renderers, this
+  // works fine. This will stop working if android_webview starts swapping out
+  // renderers on navigation.
+  bool application_initiated =
+      !is_content_initiated || type == blink::WebNavigationTypeBackForward;
+
+  // Don't offer application-initiated navigations unless it's a redirect.
+  if (application_initiated && !is_redirect)
+    return false;
+
+  bool is_main_frame = !frame->parent();
+  const GURL& gurl = request.url();
+  // For HTTP schemes, only top-level navigations can be overridden. Similarly,
+  // WebView Classic lets app override only top level about:blank navigations.
+  // So we filter out non-top about:blank navigations here.
+  if (!is_main_frame &&
+      (gurl.SchemeIs(url::kHttpScheme) || gurl.SchemeIs(url::kHttpsScheme) ||
```

Comment:
The same as WebView, there are testCalledForIframeHttpNavigations, and the test for "about:blank".